systemd/systemd.go

```diff
@@ -490,6 +499,48 @@ func (s *systemd) daemonReloadNoLock() error {
 	return err
 }
 
+type ReloadReason struct {
+	UnitsAdded   []string
+	UnitsChanged []string
+}
+
+func (s *systemd) DaemonReloadIfNeeded(reason ReloadReason) error {
+	const locked = false
+	return s.daemonReloadIfNeededWithLock(locked, reason)
+}
+
+func (s *systemd) daemonReloadIfNeededWithLock(locked bool, reason ReloadReason) error {
+	needReload := false
+	if len(reason.UnitsChanged) > 0 {
+		status, err := s.Status(reason.UnitsChanged)
+		if err != nil {
+			return err
+		}
+		// check if any unit needs reload
+		for i := 0; i < len(status); i++ {
+			if status[i].NeedDaemonReload {
+				needReload = true
+				break
+			}
+			// TODO: check if unit is gone by inspecting its
+			// Installed state, this needs to be coupled with
+			// --no-reload for disable/enable such that we do not
+			// execute the reload twice
+		}
+	} else {
+		// unconditional reload
```

Comment:
mmm... why? Did you mean to write `} else if len(reason.UnitsAdded) > 0 {` perhaps?

Reply:
We don't have a UnitsAdded field anyway, so I think this is clear enough now :-)